fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexOptions.java

```diff
@@ -110,6 +110,11 @@
      */
     public static final String BITMAP_VALUE_ENTRY_SIZE_OPTION = "bitmapValueEntrySize";
 
+    /**
+     * Whether to remove index entry for {@link IndexTypes#COUNT} type indexes when they decrement to zero.
+     */
+    public static final String CLEAR_WHEN_ZERO = "clearWhenZero";
```

Comment:
It feels like we should have an opinion about whether this index option can be changed or not (without needing to rebuild the index). From what I can tell, it's fine to change it from `false` to `true`, but you might get some zeros back on older data (unless we also add a filter in the index scan to remove zeros), and it's fine to change it from `true` to `false` except you might miss some zeros. Not sure if that's something we should mention/encourage, or if we should tell people to rebuild the index if they want to change this option. It also seems like we should document the decision in the Javadoc here.

Thoughts?

Reply:
I considered whether to override `validateChangedOptions` and concluded that it was not appropriate. Changing the option does not make the index invalid. You just have to understand that the option refers to clearing at the time of decrement to zero. And the caller is free to cause the index to be rebuilt if their requirement is that zero entries go away. There is currently nothing in record layer that makes this kind of decision.

I added a comment about this. Maybe it should go into some documentation someplace, instead?

It's like turning it on with `SUM`; it's possible that some entries will get cleared that would be `0` if rebuilt.

Reply:
Yeah, I think the contract makes sense. I think that comment gets the idea across. I do feel like we should make it available in the published Javaodoc, and I was thinking that this Javadoc here on the option would be the logical place to document it, but maybe I'm wrong.